google/earthenterprise
# Create a secure option for asset roots

Adds a `--secure` flag to `geconfigureassetroot` and `geupgradeassetroot` to remove world-writable permissions.

To test:

- Build and install.
- Verify that the fusion executable is owned by `root:gegroup` and has the following permissions: `-r-xr-sr-x`.
- Verify that both `/gevol/assets/.config` and `/gevol/assets/.userdata` both are world writable.
- Open fusion and attempt to add a provider or sever association.
- Verify that the operation succeeds.
- Repair the asset root with the secure flag: `geconfigureassetroot --repair --secure`
- Verify that both `/gevol/assets/.config` and `/gevol/assets/.userdata` both are not world writable.
- Open fusion and attempt to add a provider or server association.
- Verify that the operation succeeds.
- Run the tutorial data scripts (for good measure).
